swarm/swarm.go

```diff
@@ -353,7 +359,7 @@ func (self *Swarm) Start(srv *p2p.Server) error {
 	newaddr := self.bzz.UpdateLocalAddr([]byte(srv.Self().String()))
 	log.Info("Updated bzz local addr", "oaddr", fmt.Sprintf("%x", newaddr.OAddr), "uaddr", fmt.Sprintf("%s", newaddr.UAddr))
 	// set chequebook
-	if self.config.SwapEnabled {
+	if self.config.SwapEnabled && self.config.SwapAPI != "" {
```

Comment:
where does `swapAPI` this come from and what is it supposed to be?

Reply:
I believe it needs to be revisited at the time when we integrate chequebook contracts again, and probably we will need a **different** strategy.

But currently it would not be possible to run any tests with swap as if swap is enabled and no chequebook contract (or a fake one) is provided, then the node would just crash.

Reply:
maybe let's disable the chequebook altogether and revisit _that_ once we reach the point in which it is relevant again?